### ISSUE TITLE
Parameter coersion fails for externally bound function of type Action<string,int>

### ISSUE THREAD

**Issue by nullObjectPtr** [inkle/ink#52]
Having some trouble with external function binding. I was able to get a function of type Action<int> to work, but another one of type Action<string,int> is throwing exceptions when called. Specifically "failed to cast string to int32". There may be function signatures that will trigger this but I have not explored any other combinations.

Minimal example provided below

Cheers, 
-Jonathan

Ink File:
```
EXTERNAL grow_apples(amount)
EXTERNAL give_apples_to(id, amount)

-> Begin

== Begin ==
{grow_apples(4)}
{give_apples_to("beth", 4)}
-> DONE
```

MinInkExample.cs with compiled story inlined
```
using System;
using Ink.Runtime;

public class MinInkExample : object
{
    private string storyText = @"{""inkVersion"":11,""root"":[[{""->"":""Begin""},null],""done"",{""Begin"":[[[""G>"",[""ev"",[4,{""x()"":""grow_apples"",""exArgs"":1},null],""out"",""/ev"",null],""G<"",null],""\n"",[""G>"",[""ev"",[""str"",""^beth"",""/str"",4,{""x()"":""give_apples_to"",""exArgs"":2},null],""out"",""/ev"",null],""G<"",null],""\n"",""done"",null],null]}]}";
    public MinInkExample()
    {
        Story s = new Story(storyText);
        s.BindExternalFunction("grow_apples", new Action<int>(GrowApples));
        s.BindExternalFunction("give_apples_to", new Action<string, int>(GiveApplesTo));
        s.ContinueMaximally();
    }

    // OK
    private void GrowApples(int howMany)
    {
        string msg = "you grow " + howMany + " apples.";
    }

    // Throws: System.Exception: Failed to cast String to Int32
    private void GiveApplesTo(string who, int howMany)
    {
        string msg = who + " has " + howMany + " apples.";
    }
}
```

**Comment by Fahrengeit** [comment]
Have you tried to make howMany as String and convert it to Int? 
Maybe in function with string Ink makes the second parameter String too, and then it doesn't cast to Int. But it is just a guess

**Comment by nullObjectPtr** [comment]
I can certainly work around, but the the source of the bug could be a larger issue, say for instance, an error in the parser. I'll poke around source and see what the cause may be when I get some time later in the week.

**Comment by MattConrad** [comment]
nullObjectPtr, your params are getting applied in opposite order to what you'd expect. You're actually passing (4, "beth"). Switch your .ink to

`{give_apples_to(4, "beth")}`

you will then pass ("beth", 4) and it won't crash.

This is happening in Story.cs:CallExternalFunction(), see the Pop arguments comment. 

This is kind of confusing, maybe wants revision, or maybe just better documentation. The unit test TestExternalBinding() does show awareness of how param order is reversed.

**Comment by nullObjectPtr** [comment]
Thanks for the explanation Matt. 

I double checked the "RunningYourInk.md" for a note on this. I was wondering if I missed something.The multiply example could be re-written to make note of this.  

It looks like its would only be a small fix to add the elements to the arguments array in "reverse" order, or is it more like "double reverse order"? I can branch locally and work around it, but I'd like to politely request that this be revised, I think this will be a source of friction for most users.

Thanks again!

**Comment by joethephish** [comment]
You definitely shouldn't have to apply the arguments in reverse order! I'll look into it tomorrow, that sounds very wrong...

**Comment by joethephish** [comment]
Okay, thanks guys, that's now fixed in commit 486fef81902959be26f137ab690043d00524a941 on the development branch. I've fixed the existing test, and made the reversed the parameters so they're the right way round. Should be making it into a release build early next week.
